google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/impl/PubSubApi.scala

```diff
@@ -121,87 +128,138 @@ private[pubsub] trait PubSubApi {
       AcknowledgeRequest(json.asJsObject.fields("ackIds").convertTo[immutable.Seq[String]]: _*)
     def write(ar: AcknowledgeRequest): JsValue = JsObject("ackIds" -> ar.ackIds.toJson)
   }
-  private implicit val pullRequestFormat = DefaultJsonProtocol.jsonFormat2(PubSubApi.PullRequest)
+  private implicit val pullRequestFormat = DefaultJsonProtocol.jsonFormat2(PullRequest)
+
+  private def pool[T](hcc: Option[HttpsConnectionContext])(
+      implicit as: ActorSystem
+  ): Flow[(HttpRequest, T), (Try[HttpResponse], T), Http.HostConnectionPool] =
+    if (isEmulated) {
+      Http().cachedHostConnectionPool[T](PubSubGoogleApisHost, PubSubGoogleApisPort)
+    } else {
+      Http().cachedHostConnectionPoolHttps[T](PubSubGoogleApisHost,
+                                              PubSubGoogleApisPort,
+                                              hcc.getOrElse(Http().defaultClientHttpsContext))
+    }
 
-  def pull(project: String,
-           subscription: String,
-           maybeAccessToken: Option[String],
-           returnImmediately: Boolean,
-           maxMessages: Int)(
+  def pull[T](project: String,
+              subscription: String,
+              returnImmediately: Boolean,
+              maxMessages: Int,
+              parallelism: Int,
+              hcc: Option[HttpsConnectionContext] = None)(
       implicit as: ActorSystem,
       materializer: Materializer
-  ): Future[PullResponse] = {
+  ): Flow[(Done, Option[String], T), (Future[PullResponse], T), NotUsed] = {
```

Comment:
The code doesn't use the context in `T` right now. The `pull` implementation is only used with `subscribe`, so I think we do not need it here.
If there is a use case for passing context, the `mapAsyncUnordered` should become a `mapAsync` as it could reorder elements which `FlowWithContext` shouldn't allow.

Reply:
Yes, it is not used yet. I have a plan to use it. However, I want to limit number of changes in the first step. I'll expose it to the public API in next commit. I'll change `mapAsync` to unordered.